libs/blocks/modal/modal.js

```diff
@@ -37,6 +37,13 @@ function closeModals(modals) {
   }
 }
 
+async function decorate(block) {
+  if (block.id === 'langnav') {
+    const { default: dec } = await import('./decorators/langnav.js');
+    dec(block);
+  }
+}
+
```

Comment:
I don't think this should be here. There's better was to do this than putting lang nav logic inside our modal.

Reply:
I fixed this in my event-modal branch. You can wait for its merge.

Reply:
@auniverseaway I've merged the event-modal branch and looked at the code, and for me I still have same "issue", that my html content is being loaded in the "getPathModal" function. I can only decorate my links after that html is loaded. 
From what I can see, if we don't want to have this code in the langnav modal, I'll need to either: 
- Wrap the content in the langnav word file into a new block called langnav. I'd then need to have an event listener on the click of the change region link to load the html, decorate it, and then pass the decorated html to the getModal method. 
- Redesign how fragments are loaded into a modal, since langnav is a fragment.
- Emit an event from the modal.js file after a modal is completely loaded. I could then have a listener somewhere which knows to then decorate the langnav modal. 

All of these approaches seem quite heavy & intrusive considering the small change we want to do. Could you take another look and if you're sure there's a better way now with the event modals, then please let me know. :)

Reply:
I will fix this one way or another. We should not have other blocks living in modal.js.